src/Components/WebAssembly/Server/src/Builder/WebAssemblyRazorComponentsEndpointConventionBuilderExtensions.cs

```diff
@@ -24,6 +26,24 @@ public static class WebAssemblyRazorComponentsEndpointConventionBuilderExtension
 
         callback?.Invoke(options);
 
+        if (options.ServeMultithreadingHeaders)
+        {
+            builder.Add(endpointBuilder =>
+            {
+                var needsCoopHeaders = endpointBuilder.Metadata.OfType<ComponentTypeMetadata>().Any() // e.g., /somecomponent
+                    || endpointBuilder.Metadata.OfType<WebAssemblyRenderModeWithOptions>().Any();     // e.g., /_framework/*
+                if (needsCoopHeaders && endpointBuilder.RequestDelegate is { } originalDelegate)
+                {
+                    endpointBuilder.RequestDelegate = httpContext =>
+                    {
+                        httpContext.Response.Headers["Cross-Origin-Embedder-Policy"] = "require-corp";
+                        httpContext.Response.Headers["Cross-Origin-Opener-Policy"] = "same-origin";
```

Comment:
Can this be `same-origin-allow-popups`? If we are not having a way to configure this, we should at least be as lax as possible while enabling the feature.

Reply:
No, unfortunately that's not enough for `SharedArrayBuffer` to be enabled. It has to be `same-origin`.